### ISSUE TITLE
New hardware features

### ISSUE THREAD

**Issue by GoogleCodeExporter** [MikeBland/er9x#257]
```
What new feature/ability do you want?

I'd like to implement a tachometer.

What is the expected output?

In a new menu I want to meter the rotor rpm.

What version of the product are you using?


Please provide any additional information below.

I think is easy to use a free input port or another one from a switch in its 
open position for to connect trough a stereo mini jack the probe, or mount it 
fixed in the transmitter.
```

Original issue reported on code.google.com by `lupit...@gmail.com` on 3 Mar 2011 at 10:53

**Comment by GoogleCodeExporter** [comment]
```
Problem is...  There isn't a free input.
```

Original comment by `erezra...@gmail.com` on 3 Mar 2011 at 11:06

**Comment by GoogleCodeExporter** [comment]
```
Yes, we have some free inputs :-)
Every switch (and we have 6 keys + 4 trims + 7 switches) in its open state can 
be used as a digital input for another purpose, an open collector source can 
drive this input in parallel with the switch. 
Imagine the trainer switch in its normal position (rear position), we can use 
the same digital input for a square signal from a simple photodiode board 
(maybe we need to use a tx-rx IR diode)or more sophisticated sound detector (as 
the iPhone/Android applications).
Do you understand me? Maybe tomorrow I can publish a simple squematics.
My propose is.. if you can program it...I believe I can make the hardware for 
it.

```

Original comment by `lupit...@gmail.com` on 4 Mar 2011 at 12:31

**Comment by GoogleCodeExporter** [comment]
```
Now, that is a nice idea. a built in tachometer. It would make it easy to 
adjust the engine's mixture and also choose the right prop without having to 
buy an extra gadget. Can any other tx do this?? :-)
```

Original comment by `joaoalve...@googlemail.com` on 4 Mar 2011 at 7:46

**Comment by GoogleCodeExporter** [comment]
```
Why not use the trainer port ? It can function as an input !! 
```

Original comment by `harm.del...@echostar.com` on 4 Mar 2011 at 9:48

**Comment by GoogleCodeExporter** [comment]
```
We can use the C_0 bit or the G_2 too, they are free.

Or if we can do a major change over the project we can make the arduino 
modification (E_0 & E_1 ports changes to C_6 & C_7) and then use the same 
programmer port when there are the MISO and MOSI pins for the probes.

Then we will use a tachometer, a thermometer..., with a single 8 pins PIC or 
Atmel and communicate it with the Atmega trough the MISO/MOSI pins.

I'm working in the hardware design for a IR tachometer. The acoustic tachometer 
seems to much complex.


```

Original comment by `lupit...@gmail.com` on 4 Mar 2011 at 11:44

**Comment by GoogleCodeExporter** [comment]
```
the MISO and MOSI are already being used for FrSky telemetry. but as you guys 
say there are 2 free ports on the tx's atmega. I don't have much knowledge 
about this stuff but my guess is that if the sensor is connected to one of 
those, someone just had to adapt the FW. maybe we could even select one of the 
unsuded swithches for the model we're using.. I'm just dreaming here. as I said 
i don't understand much about this..
```

Original comment by `joaoalve...@googlemail.com` on 4 Mar 2011 at 12:26

**Comment by GoogleCodeExporter** [comment]
```
oh, and then the sensor would fit (in my case) in the slot where the original 
antenna was. how perfect would that be? just point the tx at the model's prop 
and mesure the rpm..   ;-)
```

Original comment by `joaoalve...@googlemail.com` on 4 Mar 2011 at 12:29

**Comment by GoogleCodeExporter** [comment]
Original comment by `erezra...@gmail.com` on 5 Sep 2011 at 5:48
- Changed state: **WontFix**
